### ISSUE TITLE
Mosquitto Addon doesn't work over ssl anymore

### ISSUE THREAD

**Issue by DunklerPhoenix** [home-assistant/addons#2350]
### Describe the issue you are experiencing

I have multiple devices that connect to the mqtt via port 8883 (Normal MQTT with SSL).
Since a while (2 or 3 Months) it's not possible to connect with them anymore.
The broker does not trust any client certificates anymore.

Example clients are:
Sleep as Android 
iotLink (windows)

### What type of installation are you running?

Home Assistant OS

### Which operating system are you running on?

Home Assistant Operating System

### Which add-on are you reporting an issue with?

Mosquitto broker

### What is the version of the add-on?

6.0.1

### Steps to reproduce the issue

1. Setup MQTT Addon with standard configuration
2. Connect a client over 8883
3. See the errors

### Anything in the add-on logs that might be useful for us?

```txt
1642374279: OpenSSL Error[0]: error:14094416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown
1642374279: Socket error on client <unknown>, disconnecting.
1642374281: New connection from 178.197.204.48 on port 8883.
1642374281: OpenSSL Error[0]: error:14094416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown
1642374281: Socket error on client <unknown>, disconnecting.
1642374281: New connection from 178.197.204.48 on port 8883.
1642374281: OpenSSL Error[0]: error:14094416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown
1642374281: Socket error on client <unknown>, disconnecting.
1642374289: New connection from 192.168.1.1 on port 8883.
1642374289: OpenSSL Error[0]: error:14094416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown
1642374289: Socket error on client <unknown>, disconnecting.
1642374294: New connection from 192.168.1.1 on port 8883.
1642374294: OpenSSL Error[0]: error:14094416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown
1642374294: Socket error on client <unknown>, disconnecting.
1642374304: New connection from 192.168.1.1 on port 8883.
1642374304: OpenSSL Error[0]: error:14094416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown
1642374304: Socket error on client <unknown>, disconnecting.
1642374309: New connection from 192.168.1.1 on port 8883.
1642374309: OpenSSL Error[0]: error:14094416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown
1642374309: Socket error on client <unknown>, disconnecting.
1642374318: New connection from 192.168.188.37 on port 8883.
1642374318: OpenSSL Error[0]: error:14094416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown
```

**Comment by mdegat01** [comment]
I have this addon up and running right now with multiple clients connected to it over SSL and each has their own client certificate. So this isn't universally broken, can you provide more information about your config? I can reproduce kind of similar looking errors in my own setup if I try connecting without a client certificate:
```
1642403944: New connection from 192.168.1.11 on port 8883.
1642403944: OpenSSL Error[0]: error:1417C0C7:SSL routines:tls_process_client_certificate:peer did not return a certificate
1642403944: Socket error on client <unknown>, disconnecting.
```
Or if I try connecting with a client certificate issued by a CA other then the one my broker is looking for:
```
1642403464: New connection from 192.168.1.11 on port 8883.
1642403464: OpenSSL Error[0]: error:1417C086:SSL routines:tls_process_client_certificate:certificate verify failed
1642403464: Socket error on client <unknown>, disconnecting.
```
But I'm not really sure how to reproduce that exact `alert certificate` error.

It might help if you shared your config. Also perhaps discuss your setup when it comes to the certificates. I won't ask you to share those obviously but there is a good number of setup steps to get correct when creating and using self-signed certificates from a local CA without errors. If you followed a guide for that then perhaps link to it and double check all that you did all the steps correctly?

**Comment by DunklerPhoenix** [comment]
Heho.
I didn't follow any guide. I just installed mosquito and setup sleep as android and iotlink to connect to it. It worked about a month and then it broke without any changes.
The configuration of the addon is the standard. Just reinstalled it yesterday and it's still the same. 

Also I never created a certificate. I think it was auto generated.

I will try to delete the certificate and make a new one. 
(Feedback in about 20-30 minutes)

**Comment by DunklerPhoenix** [comment]
So... I reissued a self signed certificate to HA with this guide here: https://gist.github.com/tiagofreire-pt/4920be8d03a3dfa8201c6afedd00305e and placed the files under /ssl/fullchain.pem and /ssl/privkey.pem. 
After a restart it's still throwing this error if the clients try to connect :/


```
1642407607: OpenSSL Error[0]: error:14094416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown
1642407607: Socket error on client <unknown>, disconnecting.
1642407617: New connection from 192.168.1.1 on port 8883.
1642407617: OpenSSL Error[0]: error:14094416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown
1642407617: Socket error on client <unknown>, disconnecting.
1642407623: New connection from 172.30.32.2 on port 1883.
1642407623: Socket error on client <unknown>, disconnecting.
1642407624: New connection from 192.168.1.1 on port 8883.
1642407625: OpenSSL Error[0]: error:14094416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown
1642407625: Socket error on client <unknown>, disconnecting.
1642407632: New connection from 178.197.204.48 on port 8883.
1642407632: OpenSSL Error[0]: error:14094416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown
1642407632: Socket error on client <unknown>, disconnecting.
```

Config Mosquitto:
```
certfile: fullchain.pem
customize:
  active: false
  folder: mosquitto
keyfile: privkey.pem
logins: []
require_certificate: false
```

Config Sleep as Android:
![Screenshot_20220117-092408](https://user-images.githubusercontent.com/1261305/149733868-a7a91fac-61e1-45a7-b50d-96fc45d08eb3.png)

Config iot Link:

![iotlink](https://user-images.githubusercontent.com/1261305/149734007-e549b8cb-18d4-4726-ba0a-7599d42d1cfb.png)

**Comment by mdegat01** [comment]
Ok I got it. The problem actually lies with Sleep As Android. It appears when you put `ssl://<url>` in that app in the MQTT section it is trying to use SSL V3 to talk to it. SSL V3 is ancient, it was deprecated in 2015 for security issues ([here's a timeline](https://kinsta.com/knowledgebase/tls-vs-ssl/#what) of the SSL/TLS versions if you're interested). Mosquitto must've allowed it for a bit longer then that but it must've been dropped when the addon updated the backend.

I can't figure out exactly when Mosquitto dropped SSL V3 since there's not a clear and obvious changelog for Mosquitto to point to. But the last version of this addon updated to Mosquitto V1.6.12 and as you can see in this [blog post](https://mosquitto.org/blog/2019/04/version-1-6-released/) announcing Mosquitto V1.6 they dropped support for TLS 1.0. TLS 1.0 is an update to SSL V3 that fixes its security issues so if that is dropped its pretty safe to assume SSL V3 support is also dropped.

So yea in short, I can't connect to my broker with Sleep As Android either. Every other client on my broker works fine but there's no way this app will connect with what its trying to do. To confirm yours is working properly besides this app please test the connection with [mqtt explorer](http://mqtt-explorer.com/) or Home Assistant itself.

I would recommend using the webhooks option in Sleep As Android and pointing that to HA. That way HA can directly receive the messages from your Sleep As Android app and use them. I know that works since that's what I do 😄 

If you feel you must use the MQTT option then you have two options that I can see:

1. Have Sleep As Android talk to your broker over the unencrypted port. You probably will want to point Sleep As Android to the LAN URL of HA in this case and not expose this port outside your LAN since its unencrypted. Which means you'll only receive these intents while on your wifi.
2. Report a bug to Sleep As Android and ask them to fix this feature to properly use the latest version of TLS for encrypted communications with the broker.

**Comment by DunklerPhoenix** [comment]
Ahhh very cool. I was using the custom component for sleep as android because I didn't know how webhooks in HA really work. Found out some seconds ago how xD 

Thank you really much.

Btw. I reported the problem with the old protocol to the developers of SAA.
